common/flatpak-transaction.c

```diff
@@ -2189,18 +2189,20 @@ search_for_dependency (FlatpakTransaction  *self,
 {
   g_autoptr(GPtrArray) found = g_ptr_array_new_with_free_func (g_free);
   int i;
+  g_autofree char *arch = flatpak_decomposed_dup_arch (runtime_ref);
 
   for (i = 0; remotes != NULL && remotes[i] != NULL; i++)
     {
       const char *remote = remotes[i];
       g_autoptr(GError) local_error = NULL;
       g_autoptr(FlatpakRemoteState) state = NULL;
 
-      state = flatpak_transaction_ensure_remote_state (self, FLATPAK_TRANSACTION_OPERATION_INSTALL, remote, NULL, &local_error);
+      state = flatpak_transaction_ensure_remote_state (self, FLATPAK_TRANSACTION_OPERATION_INSTALL, remote, arch, &local_error);
       if (state == NULL)
         {
           g_debug ("Can't get state for remote %s: %s", remote, local_error->message);
-          return FALSE;
+          g_error_propagate (error, g_steal_pointer (&local_error));
```

Comment:
```suggestion
          g_propagate_error (error, g_steal_pointer (&local_error));
```

Reply:
Done. This error was introduced because I manually copied the changes from my Endless working directory to my upstream one, and did it badly. I’ve just checked I didn’t introduce any other errors. The changes were tested in my Endless working directory.